docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,402 @@
+# OAuth
+
+OAuth 2.0 is used to authorize 3rd party applications to access `wire-server` resources on behalf of a Wire user.
+
+Currently, only 3rd party apps that have been implemented and approved by Wire are supported, e.g. Outlook Calendar Extension. OAuth is not open for public use.
+
+## Roles
+
+### The user (resource owner)
+
+The user is the resource owner who gives permission to the OAuth client to access parts of their resources.
+
+### 3rd party application (OAuth client)
+
+A 3rd party app is an application that is attempting to get access a resource on behalf of the user. It needs to get permission from the user in order to do so.
+
+### Resource server
+
+The resource server is the API server the 3rd party app attempts to access on behalf of the user. In our case the resource server is `wire-server`.
+
+### Authorization server
+
+The authorization server does the authentication of the user and establishes whether the user approves or denies the client's access request. In this case the authorization server is the same server as the resource server which is `wire-server`.
+
+## OAuth flow
+
+`wire-server` currently only supports the [authorization code flow](https://www.rfc-editor.org/rfc/rfc6749#section-4.1) which is optimized for confidential clients such as Outlook Calendar Extension.
+
+```{eval-rst}
+.. kroki::
+   :type: mermaid
+   :caption: OAuth 2.0 authorization code flow
+
+   sequenceDiagram
+       autonumber
+       actor U as User
+       participant C as Outlook Calendar Extension
+       participant A as Authorization Server (wire-server)
+       participant R as Resource Server (wire-server)
+
+       U->>C: Click login
+       C->>A: Authorization code request /authorize
+       A->>U: Redirect to login/authorization prompt
+       U->>A: Authenticate and consent
+       A->>C: Authorization code
+       C->>A: Authentication code + client crednetials
+       A->>A: Validate authentication code + client crednetials
+       A->>C: Access token
+       C->>R: Request a resource with access token (e.g. POST /conversations)
+       R->>R: Validate access token with public key from auth server
+       R->>C: Response
+```
+
+### Registering an OAuth client
+
+A new OAuth client can be register via the internal API of `brig` by providing an application name and a redirect URL:
+
+```curl
+  curl -s -X POST localhost:8082/i/oauth/clients \
```

Comment:
server-internal.example.com

Reply:
Do we really want to use the internal endpoint here? It's not reachable from the outside, and I assume someone wanting to add the calendar integration can't forward the internal endpoint to initiate the flow…